### PR TITLE
Fix RGL scene synchronization when FixedUpdate is called more frequently than Update

### PR DESCRIPTION
**Pull request by msz-rai**
Fix to the bug described [here](https://github.com/autowarefoundation/autoware-projects/issues/67#issuecomment-1705496973).

### CODE REVIEW

File: Assets/RGLUnityPlugin/Scripts/SceneManager.cs [hunk 1]
```diff
@@ -75,6 +75,7 @@ public enum MeshSource
         public static ITimeSource TimeSource { get; set; } = new UnityTimeSource();
 
         private int lastUpdateFrame = -1;
+        private int lastPhysicsCycle = -1;
```
Comment (prybicki) [inline Assets/RGLUnityPlugin/Scripts/SceneManager.cs:78]: I'd suggest sticking with Unity-specific names.
```suggestion
        private int lastFixedUpdateFrame = -1;
```

File: Assets/RGLUnityPlugin/Scripts/LidarSensor.cs [hunk 2]
```diff
@@ -91,6 +91,9 @@ public class LidarSensor : MonoBehaviour
         private LidarModel? validatedPreset;
         private float timer;
 
+        private int sameFramePhysicsCycleCounter = 0;
```
Comment (prybicki) [inline Assets/RGLUnityPlugin/Scripts/LidarSensor.cs:94]:
```suggestion
        private int fixedUpdatesInCurrentFrame = 0;
```

**Review by prybicki — COMMENTED** [review]
1. Please add a link to the issue explaining why this is needed, or describe in-place.
2. Consider using `fixedUpdate`-based naming, referring `physics` is IMHO to specific (it's only a apart of fixed update system).

**Review by prybicki — APPROVED** [review]

**Review by mackierx111 — APPROVED** [review]
LGTM

**Pull request merged**